helm/cluster-cloud-director/templates/helmreleases/netpol-helmrelease.yaml

```diff
@@ -36,17 +36,13 @@ spec:
   values:
     allowEgressToDNS:
       enabled: true
----
-apiVersion: source.toolkit.fluxcd.io/v1beta2
-kind: HelmRepository
-metadata:
-  name: {{ include "resource.default.name" $ }}-cluster
-  namespace: {{ $.Release.Namespace }}
-  annotations:
-    cluster.giantswarm.io/description: "{{ .Values.clusterDescription }}"
-  labels:
-    {{- include "labels.common" . | nindent 4 }}
-spec:
-  interval: 10m
-  url: https://giantswarm.github.io/cluster-catalog
-{{- end }}
+    {{- if $.Values.global.connectivity.proxy.enabled }}
+    # We collect the proxy variables from the cluster-values secret to feed them to the netpol helmRelease to create the CNP for proxy access.
+    {{- $secret := lookup "v1" "Secret" .Release.Namespace (printf "%s-cluster-values" (include "resource.default.name" .)) }}
```

Comment:
is this secret managed by `cluster-apps-operator`? or is it statically defined by the customer?

in the first case, I wonder if you'd have a race condition there.